src/compas/__init__.py

```diff
@@ -303,7 +303,26 @@ def get_bunny(localstorage=None):
         urlretrieve(url, destination)
 
         with tarfile.open(destination) as file:
-            file.extractall(localstorage)
+            def is_within_directory(directory, target):
+                
+                abs_directory = os.path.abspath(directory)
+                abs_target = os.path.abspath(target)
+            
+                prefix = os.path.commonprefix([abs_directory, abs_target])
+                
+                return prefix == abs_directory
+            
+            def safe_extract(tar, path=".", members=None, *, numeric_owner=False):
+            
+                for member in tar.getmembers():
+                    member_path = os.path.join(path, member.name)
+                    if not is_within_directory(path, member_path):
+                        raise Exception("Attempted Path Traversal in Tar File")
+            
+                tar.extractall(path, members, numeric_owner) 
```

Comment:
@TrellixVulnTeam thanks for the fix!

However this call seems to be broken. According to the [docs ](https://docs.python.org/3.10/library/tarfile.html) `numeric_owner` is forced to be a kwarg.


```suggestion
                tar.extractall(path, members, numeric_owner=numeric_owner) 
```